### DISCUSSION TITLE
simdjson conversion performance vs std::from_chars()

### DISCUSSION THREAD

**Discussion by lano1106** [simdjson/simdjson#2135] [General]
std::from_chars() introduced in C++17 as being the fastest conversion function in the west...

Are simdjson get_double(), get_int64() and get_uint64() methods in terms of std::from_chars() or is it using a SIMD based faster implementation?

**Comment by lemire** [comment]
Yes, std::from_chars() introduced in C++17 is very fast in GCC. You are correct.

The library used GCC is available standalone at
https://github.com/fastfloat/fast_float

**Comment by lemire** [comment]
More precisely, [you can examine the implementation in GCC](https://github.com/gcc-mirror/gcc/blob/9266d9fce2ca8ec704b932e8af7ab04432772c44/libstdc%2B%2B-v3/src/c%2B%2B17/fast_float/fast_float.h).

**Comment by jdmdmm** [comment]
So, may I ask, is the simdjson get_double, get_int etc implemented in terms of from_chars then? Or is it a faster simd oriented implementation?
Thanks

**Reply by lemire** [reply]
They are not semantically equivalent. In simdjson, we parse according to the JSON specification. This is not part of the C++ standard library.
